pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java

```diff
@@ -437,6 +437,12 @@ public ConsumerBuilderImpl(PulsarClientImpl client, Schema<T> schema) {
             if (conf.getAckTimeoutMillis() == 0) {
                 conf.setAckTimeoutMillis(DEFAULT_ACK_TIMEOUT_MILLIS_FOR_DEAD_LETTER);
             }
+
+            // when MaxRedeliverCount <= 0 in DeadLetterPolicy, we reset MaxRedeliverCount
+            // to default value, Default: 16.
+            if (deadLetterPolicy.getMaxRedeliverCount() <= 0) {
+                deadLetterPolicy.setMaxRedeliverCount(RetryMessageUtil.MAX_RECONSUMETIMES);
```

Comment:
It's better to add a warn log here to show users which value is applied.

And need an unit test to make sure we will not break here.

Reply:
Maybe a `checkArgument` is better here to keep consistent with other parameter validation. @codelipenghui WDYT

Reply:
Thanks @Jason918 @codelipenghui review, fix done please check again